### PR TITLE
Doc: added triage section to new contributor docs

### PR DESCRIPTION
**Pull request by story645**
## PR summary

Discussion from #28335 had me thinking that my issues w/ the guidance being in the wrong place was that triage is its own category of contribution and therefore it should get a section in `contribute.html` 

This also incorporates the guidance from #28335 (particularly language/reasoning from https://github.com/matplotlib/matplotlib/pull/28335#issuecomment-2146751622) and therefore supercedes it. 

## PR checklist

- [ ] "closes #0000" is in the body of the PR description to [link the related issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [ ] new and changed code is [tested](https://matplotlib.org/devdocs/devel/testing.html)
- [ ] *Plotting related* features are demonstrated in an [example](https://matplotlib.org/devdocs/devel/document.html#write-examples-and-tutorials)
- [ ] *New Features* and *API Changes* are noted with a [directive and release note](https://matplotlib.org/devdocs/devel/api_changes.html#announce-changes-deprecations-and-new-features)
- [ ] Documentation complies with [general](https://matplotlib.org/devdocs/devel/document.html#write-rest-pages) and [docstring](https://matplotlib.org/devdocs/devel/document.html#write-docstrings) guidelines

### CODE REVIEW

File: doc/devel/contribute.rst [hunk 1]
```diff
@@ -161,6 +161,29 @@ please reach out on the :ref:`contributor_incubator`
...
 
+.. _contribute_triage:
+
+Triage
+------
+The `issue tracker <https://github.com/matplotlib/matplotlib/issues>`_ is our
+centralized location for feature requests, bug reports,
+tracking major projects and discussing priorities. It takes a lot of
+time and effort to keep oragnized, so curating our issue tracker is a valuable
+contribution. Some examples are:
```
Comment (timhoffm) [inline doc/devel/contribute.rst:172]: We can leave out the "time and effort" part. First, it doesn't sound very attractive to embark on something "taking a lot of time and effort", even if you don't have to do everything. Second, it sounds worse than it is. While it is work, I don't have the impression it's that much, e.g. compared to PR reviewing.

```suggestion
tracking major projects and discussing priorities. Helping to keep it
organized is a valuable contribution. Some examples are:
```

Alternative wording:

```suggestion
tracking major projects and discussing priorities. Curating it is a valuable
contribution. Some examples are:
```

File: doc/devel/contribute.rst [hunk 2]
```diff
@@ -161,6 +161,29 @@ please reach out on the :ref:`contributor_incubator`
...
+
+* labeling issues and pull requests
+* verifying bug reports
+* debugging and resolving issues
+* linking to related issues, discussion, and external work
+
+Helping out with triage is discussed in detail in :ref:`bug_triaging`.
```
Comment (timhoffm) [inline doc/devel/contribute.rst:179]: As with code and documentation we should link the Contributor incubator. Code / documentation says

> If you get stuck, please reach out on the [Contributor incubator](https://matplotlib.org/devdocs/devel/contribute.html#contributor-incubator)

Likely that's not exactly the best the wording for triage. Which process do we want exactly? Is it ok that people just start to comment on issues? Or should they first contact us through the incubator.

File: doc/devel/contribute.rst [hunk 3]
```diff
@@ -161,6 +161,29 @@ please reach out on the :ref:`contributor_incubator`
...
+
+Triage help is a valuable contribution because it takes a lot of time and
+judgement to effectively curate and resolve issues; therefore we very much
+appreciate thoughtful feedback to issue authors. Please do not verbatim post
+the output of a generative AI tool (such as chat GPT) as a response to a bug
+report as we expect that the issue author has either already tried that or has a
+reason for not doing so.
```
Comment (timhoffm) [inline doc/devel/contribute.rst:186]: This is a way of working aspect of triaging. It belongs under the above linked `bug triaging` docs, not here. - My idea of "Contributing guide > Getting started" is that it should give a high-level idea of the kinds contribution and direct interested people to the relevant parts of the documentation. Specific information on how (not) to do things should be in those parts of the documentation e.g. `bug triaging`, not on the overview page.

File: doc/devel/contribute.rst (hunk 2 above)
Comment (rcomer) [reply doc/devel/contribute.rst:179]: Maybe something like

> If you have any questions about the process, please reach out on...

If someone is stuck about how to triage a particular issue, it's maybe better they just leave it for someone else to pick up (at least, that's what I do...!)

I think it's fine if people just start commenting on issues.  We are not overrun with people wanting to do this so I think it's pretty easy to keep track of what anyone new is doing.  We do not require people to ask first when making other contributions like creating PRs.

File: doc/devel/contribute.rst (hunk 3 above)
Comment (timhoffm) [reply doc/devel/contribute.rst:186]: My first impulse was: This belongs in https://matplotlib.org/devdocs/devel/triage.html. But considering the definition of triage actions on that site, replying with a solution is beyond triaging. Triaging only makes sure the issue is well laid out and sorted. It does not solve it (except for invalid or duplicate issues). Therefore, I don't see that people will respond with AI as part of triaging. To make life simple here, I claim it's out of scope for the PR and we can leave that section out.

You could add something like "Proposing solutions" to the https://matplotlib.org/devdocs/devel/triage.html. It should be made clear that
- it's optional (a triager may add it if they already exactly know how to solve this)
- proposals / recommendations should be well considered and well laid out. If you are not sure, better leave the response to somebody else. This also includes auto-generated AI responses are not acceptable.

File: doc/devel/contribute.rst [hunk 4]
```diff
@@ -158,6 +158,25 @@ please reach out on the :ref:`contributor_incubator`
...
+bug reports, tracking major projects, and discussing priorities.
+
+Some examples of what we mean by triage are:
+
+* labeling issues and pull requests
+* verifying bug reports
+* debugging and resolving issues
+* linking to related issues, discussion, and external work
+
+Helping out with triage is discussed in detail in :ref:`bug_triaging`.
```
Comment (timhoffm) [inline doc/devel/contribute.rst:176]: With my non-native English, I feel "helping out" is not that welcoming. I read it as "We have a deficiency. Can you temporarily take over a task that we currently cannot fulfill".

I suggest either

```suggestion
Helping with triage is discussed in detail in :ref:`bug_triaging`.
```
or
```suggestion
These tasks are discussed in detail in :ref:`bug_triaging`.
```
But you know the language better, so please decide whether to keep as is or change.

File: doc/devel/index.rst [hunk 5]
```diff
@@ -84,43 +84,63 @@ to contributing, we recommend that you first read our
...
       .. grid:: 1
          :gutter: 1
-         :class-row: sd-fs-5
 
          .. grid-item-card::
             :link: contribute_code
             :link-type: ref
-            :shadow: none
+            :class-card: sd-outline-{primary} sd-shadow-none
+            :class-body: sd-text-{primary}
```
Comment (timhoffm) [inline doc/devel/index.rst:119]: These stylings seem to make the cards less responsive to mouse-over. There's only a minimal size/layout change:
https://output.circle-artifacts.com/output/job/5b25daa3-7646-458d-9b4d-a2b5ab7702fb/artifacts/0/doc/build/html/devel/index.html

Previously, at least the small frame changed to purple, indicating a link.
https://matplotlib.org/devdocs/devel/index.html

**Review by timhoffm — APPROVED** [review]
I approve the content. Take or leave my comments. You may self-merge.

**Review by rcomer — APPROVED** [review]
Sorry I only just noticed you pinged for another review here.  Seems good to me :+1:

**Pull request merged**